### PR TITLE
Allow Taping crates to keep inventory

### PR DESCRIPTION
**Pull request by ALongStringOfNumbers**
## What
Allows using tape on crates to allow the crate to keep its inventory.

## Outcome
Allow taping crates so that they keep their inventory when broken.

### CODE REVIEW

**Review by LAGIdiot — CHANGES_REQUESTED** [review]
It works, but I think it needs to be more polished before it will be usable. So for now, I will block it with list of things to address or at least discuss.

- [ ] Create from JEI doesn't stack with placed and broken one (NBT for taped is left in)
- [ ] taped Create in inventory looks exactly same as un-taped one
- [ ] there is no indication either on crate that it is movable when taped
- [ ] or on tape that it can be used for taping Creates to make them movable
- [ ] already taped Crate can be taped again (consumes tape again)
- [ ] colored (sprayed with Spray can) taped Crate losses color when broken - this could be really confusing as player may have colored them specifically so that he knows what is inside
- [ ] tape is possible to make only from HV+
- [ ] taped Crates are stackable (if they have same content, would be better to make them unstackable so there is no need for nbt comparison)
- [ ] taped is not reset when Create is placed again which means it can be picked again (without spilling content)
- [ ] graphical glitch showing tapped overlay on placed Crate for split second

**Comment by ALongStringOfNumbers** [comment]
The tape being only available from HV+ is something that I realized after making this PR. However, I do knot want to do any balance changes in this PR, just introduced the feature and leave any balancing to another PR, as it might touch more than just the tape.

I could not reproduce the graphical glitch that you mentioned at all

**Comment by LAGIdiot** [comment]
> The tape being only available from HV+ is something that I realized after making this PR. However, I do not want to do any balance changes in this PR, just introduced the feature and leave any balancing to another PR, as it might touch more than just the tape.

I think this is wrong approach. As we may not get to it that soon which would mean that this feature is not so useful. It would be a different story, if there was some array of items which can be used to tap crates, and we registered only tape but anyone (mostly modpack devs) could register other items from get on.

**Comment by ghzdude** [comment]
> The tape being only available from HV+ is something that I realized after making this PR. However, I do not want to do any balance changes in this PR, just introduced the feature and leave any balancing to another PR, as it might touch more than just the tape.

It's possible to make duct tape as early as LV? The earliest recipe (as of 2.5.4) just involves carbon and polyethylene, both of which are do-able in LV, unless there's something I'm missing.

**Comment by LAGIdiot** [comment]
> > The tape being only available from HV+ is something that I realized after making this PR. However, I do not want to do any balance changes in this PR, just introduced the feature and leave any balancing to another PR, as it might touch more than just the tape.
> 
> It's possible to make duct tape as early as LV? The earliest recipe (as of 2.5.4) just involves carbon and polyethylene, both of which are do-able in LV, unless there's something I'm missing.

I may have missed the Polyethylene only chain, and I see now if you use cracking in Chemical reactor you can distill Ethylene in LV. In that case it is available early enough, but to find this chain is not easy. It would be way easier if there was some early type just for this, something from glue, paper and wood pulps, maybe?

File: src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityCrate.java [hunk 1]
```diff
@@ -31,11 +36,16 @@
...
 import java.util.List;
 
+import static gregtech.api.capability.GregtechDataCodes.IS_TAPED;
+import static gregtech.api.capability.GregtechDataCodes.TAG_KEY_PAINTING_COLOR;
+
 public class MetaTileEntityCrate extends MetaTileEntity {
 
     private final Material material;
     private final int inventorySize;
     private ItemStackHandler inventory;
```
Comment (DStrand1) [inline src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityCrate.java:46]: Can this field be made `protected`? It's useful for addons (TJCore) that are trying to extend this class

**Comment by planetme** [comment]
real

**Review by serenibyss — APPROVED** [review]

**Pull request merged**